rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMRestClient.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Provider;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.codehaus.jackson.map.DeserializationConfig;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.jboss.resteasy.client.jaxrs.ClientHttpEngine;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
+import org.jboss.resteasy.client.jaxrs.engines.ApacheHttpClient4Engine;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.ow2.proactive_grid_cloud_portal.common.RMRestInterface;
+import org.ow2.proactive_grid_cloud_portal.common.exceptionmapper.ExceptionToJson;
+
+
+public class RMRestClient {
+
+    private RMRestInterface rm;
+
+    private String restEndpointURL;
+
+    private ResteasyProviderFactory providerFactory;
+
+    private static ClientHttpEngine httpEngine;
+
+    private static SSLContext sslContext;
+
+    public RMRestClient(String restEndpointURL, ClientHttpEngine httpEngine) {
+
+        if (httpEngine == null) {
+            setBlindTrustSSLContext();
+            SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(sslContext,
+                                                                                                   NoopHostnameVerifier.INSTANCE);
+            HttpClient httpClient = HttpClientBuilder.create().setSSLSocketFactory(sslConnectionSocketFactory).build();
+            httpEngine = new ApacheHttpClient4Engine(httpClient);
+        }
+
+        this.httpEngine = httpEngine;
+        this.restEndpointURL = restEndpointURL;
+
+        providerFactory = ResteasyProviderFactory.getInstance();
+        if (!providerFactory.isRegistered(RMRestClient.JacksonContextResolver.class)) {
+            providerFactory.registerProvider(RMRestClient.JacksonContextResolver.class);
+        }
+        //        registerGzipEncoding(providerFactory);
```

Comment:
Explain why you keep this code commented or remove it please.

Reply:
thanks

Reply:
Identified issue is still there.